sycl/include/sycl/ext/oneapi/experimental/builtins.hpp

```diff
@@ -117,11 +123,154 @@ inline __SYCL_ALWAYS_INLINE
 
 } // namespace native
 
-} // namespace experimental
-} // namespace oneapi
-} // namespace ext
+template <typename T>
+std::enable_if_t<std::is_same<T, bfloat16>::value, T> fabs(T x) {
```

Comment:
If I remove `Template <typename T>` and usage of `enable_if_t` then the compiler sees multiple definitions of `bfloat16 fabs()` with the same uint16_t (bfloat16 storage type) mangled name. I'm not completely sure why this is or why the templating and use of `enable_if_t` resolved this but I guessed it gets confused with the other marray definition.

Reply:
Ah! It's probably confused by some ambiguity between `bfloat16` and it's storage class, thinking this could be called if passed `uint16_t` through implicit conversion. Interesting! Thank you for clarifying. 😄